service/src/main/java/tds/assessment/repositories/impl/AssessmentMapper.java

```diff
@@ -20,51 +21,24 @@
     /**
      * Maps the results to an {@link tds.assessment.Assessment}
      *
-     * @param rows
-     * @return
+     * @param rows data result rows that represent a result row from the query result
+     * @return optional with {@link tds.assessment.Assessment} otherwise empty
      */
     Optional<Assessment> mapResults(List<Map<String, Object>> rows) {
-        Map<String, SegmentInformationHolder> segments = new LinkedHashMap<>();
-        Map<String, Object> assessmentRow = null;
-        for (Map<String, Object> row : rows) {
-            if (row.get("assessmentKey") == null) { // This is the assessment row
+        Map<String, Segment> segments = new LinkedHashMap<>();
+        ResultRow assessmentRow = null;
+        for (Map<String, Object> result : rows) {
+            ResultRow row = new ResultRow(result);
+            if (row.getString("assessmentKey") == null) { // This is the assessment row
                 assessmentRow = row;
             } else {  // This is a segment row
-                String key = (String) row.get("assessmentSegmentKey");
+                String key = row.getString("assessmentSegmentKey");
 
                 if (!segments.containsKey(key)) {
-                    segments.put(key, new SegmentInformationHolder());
-                }
-
-                SegmentInformationHolder holder = segments.get(key);
-                String segmentKey = (String) row.get("assessmentSegmentKey");
-
-                //If holder is null this is the first record for this segment
-                Segment segment = new Segment(segmentKey);
```

Comment:
This block is mostly the same between the segmented assessment parsing and a non segmented assessment parsing so pull it into a private method below.